src/main/java/bio/terra/service/snapshot/SnapshotStorageAccountDao.java

```
@@ -0,0 +1,90 @@
+
+  private static final String sqlCreateLink =
+      "UPDATE snapshot "
+          + " SET storage_account_resource_id = :storage_account_resource_id"
+          + whereClause;
+
+  private static final String sqlGetStorageAccountResourceId =
+      "SELECT storage_account_resource_id FROM snapshot"
+          + whereClause
+          + " AND storage_account_resource_id IS NOT NULL";
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public SnapshotStorageAccountDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public void createSnapshotStorageAccountLink(UUID snapshotId, UUID storageAccountResourceId) {
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("snapshot_id", snapshotId)
+            .addValue("storage_account_resource_id", storageAccountResourceId);
+
+    try {
+      jdbcTemplate.update(sqlCreateLink, params);
+    } catch (DataAccessException dataAccessException) {
+      if (retryQuery(dataAccessException)) {
+        logger.error("snapshotStorageAccount link operation failed with retryable exception.");
+        throw new RetryQueryException("Retry", dataAccessException);
+      }
+      logger.error("snapshotStorageAccount link operation failed with fatal exception.");
+      throw dataAccessException;
+    }
+  }
+
+  public Optional<UUID> getStorageAccountResourceIdForSnapshotId(UUID snapshotId) {
+    MapSqlParameterSource params = new MapSqlParameterSource().addValue("snapshot_id", snapshotId);
+
+    List<UUID> storageAccountResourceIds =
+        jdbcTemplate.query(
+            sqlGetStorageAccountResourceId, params, new UuidMapper("storage_account_resource_id"));
+
+    if (!storageAccountResourceIds.isEmpty()) {
+      return Optional.of(storageAccountResourceIds.get(0));
```

Should we throw an error here if there's more than one result?